simulation/groups.py

```diff
@@ -6,337 +6,186 @@
 ]
 
 from simulation import groups_interface
-import random
+import math
 
 
 class SGroup(groups_interface.PopulationGroup):
-    def __init__(self, x: float, y: float, label: str, inf_dist: float, infection_prob: float, recover_prob: float,
-                 death_prob: float) -> None:
-
-        super().__init__(x, y, label, inf_dist, infection_prob, recover_prob, death_prob)
-
-    def __del__(self):
-        pass
-
-    def move(self, x_distance: float, y_distance: float, box_width: int,
-             box_height: int) -> bool:
-
-        can_move = True
-
-        if not 0 < self.x + x_distance < box_width:
-            can_move = False
-
-        if not 0 < self.y + y_distance < box_height:
-            can_move = False
-
-        if can_move:
-            self.x += x_distance
-            self.y += y_distance
-
-            return True
-        else:
-            return False
-
-    def is_in_area(self, member_x: float, member_y: float, member_inf_dis: float) -> bool:
-        if self.x > member_x:
-            x_difference = self.x - member_x
-        else:
-            x_difference = member_x - self.x
-
-        if self.y > member_y:
-            y_difference = self.y - member_y
-        else:
-            y_difference = member_y - self.y
-
-        if x_difference < member_inf_dis or \
-                y_difference < member_inf_dis:
-            return True
-        else:
-            return False
-
-    def infect(self):
-        x = random.random()
-
-        if x < self.infection_probability:
-            return True
-        else:
-            return False
-
-    def recover(self):
-        x = random.random()
-
-        if x < self.recover_probability:
-            return True
-        else:
-            return False
-
-    def death(self):
-        x = random.random()
-
-        if x < self.death_probability:
-            return True
-        else:
-            return False
-
-    # Documentation inheritance handling
-    __doc__ = groups_interface.PopulationGroup.__doc__
-    move.__doc__ = groups_interface.PopulationGroup.move.__doc__
-    is_in_area.__doc__ = groups_interface.PopulationGroup.is_in_area.__doc__
-    infect.__doc__ = groups_interface.PopulationGroup.infect.__doc__
-    recover.__doc__ = groups_interface.PopulationGroup.recover.__doc__
-    death.__doc__ = groups_interface.PopulationGroup.death.__doc__
+    """
+    Class that represents susceptible people that can be infected. Has own
+    behaviour method infect() which define if person get infected or not.
+
+    Parameters
+    ----------
+    infection_prob: float, required
+        Probability to infect it should be value between 0.0 and 1.0.
+    **kwargs:
```

Comment:
I do not think this is the correct way of documenting parameters but I may be wrong.
From my perspective it is not an elegant way (still better to be explicit than implicit and pass / document all interface)